src/index.ts

```diff
@@ -254,9 +268,11 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
 
     const defaultOptions: PoolOptionsBase = {
       poolSize: 10,
+      minPoolSize: 0,
       idleTimeoutMillis: 10000,
       waitForAvailableConnectionTimeoutMillis: 90000,
-      connectionTimeoutMillis: 30000,
+      connectionTimeoutMillis: options.minPoolSize && options.minPoolSize > 0 ? 0 : 30000,
```

Comment:
I think this should stay at a default of 30s, regardless if there is a minimum pool size